Data/Pool.hs

```diff
@@ -175,26 +178,27 @@ createPool'
     -- longer than requested, as the reaper thread wakes at 1-second
     -- intervals.
     -> Int
-    -- ^ Maximum number of resources to keep open per stripe.  The
-    -- smallest acceptable value is 1.
+    -- ^ Initial maximum number of resources to keep open per stripe.
+    -- The smallest acceptable value is 1.
     --
     -- Requests for resources will block if this limit is reached on a
     -- single stripe, even if other stripes have idle resources
     -- available.
     -> Maybe NominalDiffTime
     -- ^ Amount of time to wait while attempting to acquire a resource.
     -> IO (Pool a)
-createPool' create destroy numStripes idleTime maxResources timeout = do
+createPool' create destroy numStripes idleTime maxResources' timeout = do
   when (numStripes < 1) $
     modError "pool " $ "invalid stripe count " ++ show numStripes
   when (idleTime < 0.5) $
     modError "pool " $ "invalid idle time " ++ show idleTime
-  when (maxResources < 1) $
-    modError "pool " $ "invalid maximum resource count " ++ show maxResources
+  when (maxResources' < 1) $
+    modError "pool " $ "invalid maximum resource count " ++ show maxResources'
   localPools <- V.replicateM numStripes $
                 liftM3 LocalPool (newTVarIO 0) (newTVarIO []) (newIORef ())
   reaperId <- forkIOLabeledWithUnmask "resource-pool: reaper" $ \unmask ->
                 unmask $ reaper destroy idleTime localPools
+  maxResources <- newTVarIO maxResources'
   fin <- newIORef ()
   let p = Pool {
```

Comment:
nit: the `p` is not used anywhere else other than the `return`. We can avoid this let block.

Reply:
Resolved by using record wildcards